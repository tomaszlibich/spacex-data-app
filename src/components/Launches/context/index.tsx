import { createContext } from "react";
import { LaunchesContextType } from "../types";

export const initialData: LaunchesContextType = {
  launches: [],
  sorting: null,
  setSorting: () => {},
  search: undefined,
  setSearch: () => {},
};

export const LaunchesContext = createContext(initialData);
