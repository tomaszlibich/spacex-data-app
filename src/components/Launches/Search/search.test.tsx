import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, mount } from 'enzyme';
import { LaunchesContext, initialData } from "../context";
import { LaunchesSearch } from "./index";

configure({ adapter: new Adapter() });

const setSearchSpy = jest.fn();
const initialContextData = { ...initialData, setSearch: setSearchSpy };

let component;
let tree: any;
let wrapper: any;

describe("LaunchesSearch", () => {
  describe('When component is rendered', () => {
    beforeEach(() => {
      component = renderer.create(
        <LaunchesSearch />,
      );
      tree = component.toJSON();
    });

    it("should match the snapshot", () => {
      expect(tree).toMatchSnapshot();
    });

    describe('And no search phrase has been provided', () => {
      beforeEach(() => {
        wrapper = mount(
          <LaunchesContext.Provider value={{ ...initialContextData }}>
            <LaunchesSearch />
          </LaunchesContext.Provider>
        )
      });

      it("should render only one item on the list", async () => {
        expect(wrapper.find('input').value).not.toBeDefined();
      });

      describe('And search phrase has been provided', () => {
        beforeEach(() => {
          wrapper = mount(
            <LaunchesContext.Provider value={{ ...initialContextData }}>
              <LaunchesSearch />
            </LaunchesContext.Provider>
          )

          wrapper.find('input').simulate('change', { target: { name: "search", value: "xyz" } })
        });

        it("should update the context with search phrase", async () => {
          expect(setSearchSpy).toHaveBeenCalledWith('xyz');
        });
      });
    });
  });
});
