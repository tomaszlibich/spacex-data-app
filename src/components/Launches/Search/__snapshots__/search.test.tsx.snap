// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchesSearch When component is rendered should match the snapshot 1`] = `
<div
  className="search"
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search..."
    type="text"
  />
</div>
`;
